### PR TITLE
Japanese util refactor

### PR DESCRIPTION
**Pull request by toasted-nutbread**
This PR merges the two japanese.js files into a single file. The wanakana dependency is now optional and will throw a meaningful error if any methods try to use it when it's not present.

#413, #463

### CODE REVIEW

File: ext/mixed/js/japanese.js [hunk 1]
```diff
@@ -89,63 +139,410 @@ const jp = (() => {
...
-                return false;
+        // Character code testing functions
+
+        isCodePointKanji(codePoint) {
+            return isCodePointInRanges(codePoint, CJK_UNIFIED_IDEOGRAPHS_RANGES);
+        }
+
+        isCodePointKana(codePoint) {
+            return isCodePointInRanges(codePoint, KANA_RANGES);
+        }
```
Comment (siikamiika) [inline ext/mixed/js/japanese.js:168]: This is not used anymore since `isStringEntirelyKana` uses `isCodePointInRanges` directly

Comment (toasted-nutbread) [reply ext/mixed/js/japanese.js:168]: True, I didn't notice that. I think it's probably fine to leave in though, since it's so simple.

**Comment by toasted-nutbread** [comment]
I just realized, CI didn't run for this change... 🤔

**Comment by siikamiika** [comment]
> I just realized, CI didn't run for this change... 🤔

You can try to rebase, I merged a PR

**Pull request merged**